Cart Flow/Custom Views/Cells/ListCell.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  ListCell.swift
+//  Cart Flow
+//
+//  Created by Scott Quintana on 8/3/20.
+//  Copyright © 2020 Scott Quintana. All rights reserved.
+//
+
+import UIKit
+
+class ListCell: UITableViewCell {
+    
+    static let reuseID = "ListCell"
+    
+    let listNameLabel = CFTitleLabel(textAlignment: .left, fontSize: 26)
+    let listDateUpdated = CFSecondaryTitleLabel(textAlignment: .right)
+    
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        configure()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func set(shoppingList: ShoppingList) {
+        let dateFormatter = DateFormatter()
```

Comment:
Here's that same `DateFormatter` again :P. Maybe this should be moved to a static or constant that can be accessed throughout the app. Maybe you could add it as an extension or to your constants file.

Reply:
Would it be any less expensive to set it up as an extension like this?

`extension Date {
    
    func convertToMonthDayYearFormat() -> String {
        let dateFormatter = DateFormatter()
        dateFormatter.dateFormat = "MM/dd/yyyy"
        return dateFormatter.string(from: self)
    }
}`

and then call lastPurchasedDate.convertToMonthDayYearFormat() ?
or will that still instantiate it with every cell?

Reply:
Your proposed function encapsulates this logic well, but the `DateFormatter` will still be instantiated each time you call this (you can put a breakpoint on that `let dateFormatter = DateFormatter()` call and see it'll be hit multiple times if you're showing multiple dates). Better to move the `dateFormatter` into a static var, like this:

`
class DateHelper {
    static var monthDayYearDateFormatter: DateFormatter = {
            let formatter = DateFormatter()
            formatter.dateFormat = "MM/dd/yyyy"
            return formatter
        }()

    static func convertToMonthDayYearFormat() -> String {
        return monthDayYearDateFormatter.string(from: self)
    }
}
`

Then just call `DateHelper.convertToMonthDayYearFormat(Date)` whenever you need to format a date to a string. The `monthDayYearDateFormatter` initializer will only be called once and will hold onto the `DateFormatter`, which you can access wherever you need it.

Reply:
Got this working! You did something that I've seen done before and been wondering about. After the closing bracket on the computed property you have the (), and I noticed that I got errors without it. I haven't been able to find much documentation on that specific concept.

Reply:
Ah, finally found some documentation in the official Swift docs: https://docs.swift.org/swift-book/LanguageGuide/Initialization.html#ID232 if you're curious.